libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -197,6 +197,9 @@ void NavEKF3_core::updateStateIndexLim()
 // Set inertial navigation aiding mode
 void NavEKF3_core::setAidingMode()
 {
+    resetDataSource posResetSource = resetDataSource::DEFAULT;
```

Comment:
this is a change in behaviour for the AID_NONE and AID_RELATIVE cases, it will always call the reset fns with DEFAULT after this PR, whereas would have called with previous source

Reply:
Thank you @tridge 
In current code resetDataSource  will be reset to DEFAULT in the end of ResetPosition
https://github.com/ArduPilot/ardupilot/blob/3617e0c729aeef858b7500b1444a9fc5c8493756/libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp#L159-L160

Reply:
Yes, we noticed this later in the discussion so Tridge and I are reasonably sure that it's OK but we'd still like PaulR to check.

Reply:
When AID_NONE the end result after a reset should be zero horizontal velocity and position states. When AID_RELATIVE, position should be at last known and velocity should be reset using optical flow and height above ground if available. If not available, then reset should be delayed until it is available or timeout to AID_NONE.

Reply:
How has this been tested?

Reply:
Thank you @priseboroigh

> How has this been tested?

I tested it with ext nav pos and vel